File: TicTacToe/TicTacToe/Judge.vb
Public Class Judge
    Private board As GameBoard

    Public Sub New(ByRef board As GameBoard)
        Me.board = board
    End Sub

    Public Function JudgeGame(ByRef p As IPlay) As Boolean

        If board.IsBoardEmpty() Then
            Return True
        End If

        If board.IsWin(p.GetHand()) Then
            board.Winner = p
            Return True
        End If

        Return False
    End Function

End Class

File: TicTacToe/TicTacToe/Judge.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TicTacToe
{
    class Judge
    {
        private GameBoard _board;

        public Judge(GameBoard board)
        {
            this._board = board;
        }

        internal bool judgegame(Player mainPlayer)
        {
            if (_board.IsWInner(mainPlayer))
                return true;

            if (_board.IsFinished)
                return true;

            return false;
        }
    }
}
